### PR TITLE
.clang-format: Update to match coding conventions

### LINKED ISSUE #975: Update `.clang-format`   (and `.checkpatch.conf`) according to the coding convention.
### Feature request summary

https://github.com/unikraft/docs/pull/248 describes the coding convention for Unikraft. However, the current provided `.clang-format`  (and `.checkpath.conf`) is outdated and does not reflect the current coding style. 
Supporting auto-formatting tools like clang-format would make staying consistent with the coding styles easier.

### Describe alternatives

_No response_

### Related architectures

None

### Related platforms

None

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tbunch1**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [x] Updated relevant documentation.

### Base target

 - Architecture(s): [N/A]
 - Platform(s): [N/A]
 - Application(s): [N/A]

### Additional configuration
N/A

### Description of changes
Updates .clang-format to better reflect the Unikraft coding conventions. 
Co-authored-by: Eddie Cazares <ecazares15@utexas.edu>
Co-authored-by: Lindsey Bowen <lindseyb803@gmail.com>

GitHub-Fixes: #975

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -3,9 +3,15 @@
...
+BreakBeforeBinaryOperators: None
+ColumnLimit: 80
 AllowShortFunctionsOnASingleLine: Empty
 AllowShortIfStatementsOnASingleLine: false
 IndentCaseLabels: false
+IndentGotoLabels: false
+RemoveBracesLLVM: true
 SortIncludes: false
+SpaceBeforeParens: ControlStatements
+TabWidth: 8
```
Comment (craciunoiuc) [inline .clang-format]:
```suggestion
TabWidth: 8

```

Missing newline at the end :sweat_smile:

**Comment by craciunoiuc** [comment]
I think you should also merge all commits into a single one 🙏

**Comment by tbunch1** [comment]
Apologies if this is a dumb question, but how should we do that? We're pretty new to contributing to open source and we can't figure out how to merge them together. I've been trying to rebase them but I'm having trouble because the changes are pushed. Again sorry if this is a simple question, but we're stuck

**Comment by craciunoiuc** [comment]
No worries,

on your local machine run these commands:
1. `git rebase --interactive HEAD~3`
2. in the prompt change the last two `pick` to `fixup` and the first `pick` to `reword`
3. change the first commit message to: `.clang-format: Update to match coding conventions` and continue
4. `git push --force`

I hope this works/helps 😅

**Comment by tbunch1** [comment]
Oh I see. I was worried about messing with the commit that merged the recent updates but I guess I didn't understand how it worked.

**Review by StefanJum — CHANGES_REQUESTED** [review]
@tbunch1 please remove the changes to `pagefault64.c`, since they are not part of this pr

**Comment by razvand** [comment]
> @tbunch1 please remove the changes to `pagefault64.c`, since they are not part of this pr

@tbunch1 , could you please see the comments from @StefanJum and update the PR accordingly?

See also the `checkpatch` checks.

**Review by craciunoiuc — APPROVED** [review]
I guess it looks okay. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Review by StefanJum — APPROVED** [review]
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**